src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EnumerableExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class EnumerableExtensions
+    {
+        public static IEnumerable<T> Shuffle<T>(this IList<T> values)

```

Comment:
Seems like it should return a List, and be called ListExtensions

Reply:
It should be called ListExtensions, but shouldn't we return an IEnumerable?

Reply:
If it takes a List, why downgrade to IEnumerable? I'd go with `IEnumerable<T>` in, `IEnumerable<T>` out. `List<T>` in, `List<T>` out.

Reply:
Okay, but if go with `List<T>` in, `List<T>` out, no need to have returned value. we can do an in-place shuffling, but to use the chain call, I still return it.